src/chi-vue/src/components/data-table/DataTable.tsx

```diff
@@ -681,7 +722,21 @@ export default class DataTable extends Vue {
       const isSelected = (row: DataTableRow) => this.selectedRows.includes(row.rowId);
 
       if (this.slicedData) {
-        selectAllCheckbox.checked = this.slicedData.every(isSelected);
+        if (this.slicedData.every(isSelected)) {
+          selectAllCheckbox.checked = true;
+          selectAllCheckbox.indeterminate = false;
+          selectAllCheckbox.classList.remove(INDETERMINATE_CLASS);
+          return;
+        } else if (this.slicedData.some(isSelected) && !this.slicedData.every(isSelected)) {
```

Comment:
There's no need of the second check
```suggestion
        } else if (this.slicedData.some(isSelected)) {
```